packages/components/src/input-control/reducer/reducer.ts

```diff
@@ -71,6 +71,11 @@ function inputControlStateReducer(
 	composedStateReducers: StateReducer
 ): StateReducer {
 	return ( state, action ) => {
+		// Update actions merely merge state and return without further ado.
+		if ( action.type === actions.UPDATE ) {
+			return { ...state, ...action.payload };
+		}
```

Comment:
While the rest of the commit is mostly just reorganizing this is an important actual change. In the scope of this commit, it is to avoid making a specific replacement for `setIsFocused`. In the larger scope, it provides a way to update state properties other than `value`. At present, other properties like `isDragEnabled` and `isPressEnterToChange` are fixed upon component initialization. While so far that has not been problematic it doesn't seem like it is intended to be that way.

Finally, because the only place the `UPDATE` action is currently in use is inside of `InputControl` and the more general purpose it is now intended for, returning early after the update avoids running further reducers that should never need to specialize the action.